Unable to process .zip files using Windows OS

Dear Tracespace View team,

We are unable to generate a proper output whenever a `.zip` file is used - could you please advise?
![image](https://user-images.githubusercontent.com/35899441/55875120-84a1e280-5b94-11e9-8cb7-6fdee6547388.png)

###### Notes
1. Using Windows OS with Chrome and Firefox browsers = same error code is encountered
 - We tried to use other .zip files = same error code is encountered
 - Unzipping the files and adding them separately works
2. Using MacOS = it works (no problems, so far)

###### Steps to re-create issue
1. Download test `.zip` file from `https://tracespace.io/view/arduino-uno.zip`
2. Click on the "Add file" button
3. Upload the `arduino-uno.zip` file
4. The `react-dom.production.min.js:102 Error: <use> attribute x: Expected length, "NaN".` error is encountered (See `Console Error Log` section for full reference)

###### Console Error Log
```
react-dom.production.min.js:102 Error: <use> attribute x: Expected length, "NaN".
nr	@	react-dom.production.min.js:102
Ha	@	react-dom.production.min.js:227
Qa	@	react-dom.production.min.js:232
Ka	@	react-dom.production.min.js:233
zo	@	react-dom.production.min.js:249
Po	@	react-dom.production.min.js:248
wo	@	react-dom.production.min.js:245
Ga	@	react-dom.production.min.js:243
oi	@	react-dom.production.min.js:157
(anonymous)	@	store.ts:20
l.current.onmessage	@	store.ts:56
```

--- a/apps/view/src/RenderWorker/files.ts
+++ b/apps/view/src/RenderWorker/files.ts
@@ -5,13 +5,17 @@ import pump from 'pump'
 import {baseName, promiseFlatMap, PromiseArray} from './util'
 
 const READER_OPTIONS = {chunkSize: 2048}
-const MIMETYPE_ZIP = 'application/zip'
+const MIMETYPE_ZIP = [
+  'application/zip',
+  'application/x-zip',
+  'application/x-zip-compressed',
+]
 
 export type FileToWrite = {name: string; contents: string}
 
 export async function readFiles(files: Array<File>): PromiseArray<FileStream> {
   return promiseFlatMap(files, (file: File) =>
-    file.type === MIMETYPE_ZIP ? zipReader(file) : [fileReader(file)]
+    isZip(file) ? zipReader(file) : [fileReader(file)]
   )
 }
 
@@ -22,7 +26,7 @@ export async function fetchZipFile(url: string): PromiseArray<FileStream> {
       throw new Error(`Could not fetch ${url}: ${response.status}`)
     })
     .then(blob => {
-      if (blob.type === MIMETYPE_ZIP) return zipReader(blob)
+      if (isZip(blob)) return zipReader(blob)
       throw new Error(`${url} is not a zip file`)
     })
 }
@@ -62,6 +66,13 @@ export class FileStream extends Transform {
     this.contents = Buffer.concat(this._chunks)
     next()
   }
+}
+
+function isZip(file: File | Blob): boolean {
+  return (
+    ('name' in file && file.name.endsWith('.zip')) ||
+    MIMETYPE_ZIP.includes(file.type)
+  )
 }
 
 function fileReader(file: File): FileStream {

